### ISSUE TITLE
Theme Test Data package for WP-CLI

### ISSUE THREAD

**Issue by GaryJones** [WPTRT/theme-test-data#2]
> @GaryJones asked on Twitter about the status on the different Theme Test Data.

To clarify in more than 140 characters, my question was leading on to the fact that there were at least three forked versions available to the community, none of which appeared to have been updated anytime recently. It's difficult to do a diff between them so where might have made fixes / improvements to one part, another might have improved a different part. The efforts have splintered the potential usefulness.

Having one main test data project in the community, which includes the basic markup, but also things like WooCommerce / EDD items, Gravity Forms form, and other relevant entries from the popular plugins, allowing them to be easily styled if the relevant plugin is active, would be useful.

Having a tool which would allow you to add this in a per module basis, and remove it on a per module basis would be even sweeter. An example building on the earlier suggestion would be something like:

```sh
# Initial adding in of selective test data
wp themetestdata add posts,pages,woocommerce,gravityforms
```
```sh
# Decided to switch to EDD instead
wp themetestdata remove woocommerce
wp themetestdata add edd
```
```sh
# Living on the edge
wp themetestdata remove posts,pages
wp themetestdata add gutenberg-posts,gutenberg-pages
```

```sh
# Get all data as an xml file, for folks that want to import that
wp themetestdata --all export
```

The current exported XML from WP has issues (IIRC) making manual editing a pain. Having a WP-CLI package (even if it wraps the `posts` and `export` behaviours) would seem to be a more forward-thinking and interactively usable solution than outputting a difficult to read XML file as the primary source of theme test data.

Plugin vendors could submit PRs to add in test data for their plugin, or build separate packages that `wp themetestdata` could also read from.

cc @krogsgard

**Comment by grappler** [comment]
I totally agree, what I am not fully sure, in what format the data should be saved in.

I think the content elements like tables etc could be placed in single files but information like featured image, post status etc would need to be defined is some type of config.

I am not sure if PHP or JSON would better?

**Comment by emiluzelac** [comment]
Definitely great idea to add test data for the plugins, this will be
extremely useful and much needed.

As far as the format, personally I would vote PHP.

**Comment by krogsgard** [comment]
FWIW I have scheduled a friend/contractor to help reformat WP Test data to update the standard data and tooling around it (primarily WP CLI helpers) to make it modern and useful -- slated to start in a few weeks.

Support for various types of content (Gravity Forms, WooCommerce) are definitely a priority for it.

Another potential feature will be self cleanup options for when the tester is done.

**Comment by GaryJones** [comment]
@krogsgard Is that something that can / should be folded into TRT efforts (or encourage the TRT to switch to WP Test solution)? 

Both teams tackling this at the same time seems...redundant if the end result isn't a best of both worlds that can satisfy everyone's needs.

**Comment by joyously** [comment]
There is also work on improving the export:[Feature Plugin Proposal](https://docs.google.com/document/d/18RkGb3elkvq9GHcfQt0KEguPruIu7EbrM6U_e1JFjgc/edit#heading=h.322alped3eob)

**Comment by GaryJones** [comment]
@krogsgard Did you contractor make a start? Is this something that we can see / potentially contribute to?

**Comment by carolinan** [comment]
@krogsgard
Any update?
People are submitting issues regarding the current theme unit test files and we need to decide if we should adress them or start working on this new idea.
